zebra-consensus/src/transaction.rs

```diff
@@ -153,175 +161,243 @@ where
+                    .ready_and()
+                    .await?
+                    .call((spend.rk, spend.spend_auth_sig, &shielded_sighash).into());
+
+                async_checks.push(rsp.boxed());
+            }
+
+            for output in sapling_shielded_data.outputs() {
+                // Consensus rule: cv and wpk MUST NOT be of small
+                // order, i.e. [h_J]cv MUST NOT be 𝒪_J and [h_J]wpk
+                // MUST NOT be 𝒪_J.
+                //
+                // https://zips.z.cash/protocol/protocol.pdf#outputdesc
+                check::output_cv_epk_not_small_order(output)?;
+
+                // Consensus rule: The proof π_ZKOutput MUST be
+                // valid given a primary input formed from the other
+                // fields except C^enc and C^out.
+                //
+                // Queue the verification of the Groth16 output
+                // proof for each Output description while adding
+                // the resulting future to our collection of async
+                // checks that (at a minimum) must pass for the
+                // transaction to verify.
+                let output_rsp = output_verifier
+                    .ready_and()
+                    .await?
+                    .call(primitives::groth16::ItemWrapper::from(output).into());
+
+                async_checks.push(output_rsp.boxed());
+            }
+
+            let bvk = sapling_shielded_data.binding_verification_key();
+
+            // TODO: enable async verification and remove this block - #1939
+            {
+                let item: zebra_chain::primitives::redjubjub::batch::Item =
+                    (bvk, sapling_shielded_data.binding_sig, &shielded_sighash).into();
+                item.verify_single().unwrap_or_else(|binding_sig_error| {
+                    let binding_sig_error = binding_sig_error.to_string();
+                    tracing::warn!(%binding_sig_error, "ignoring");
+                    metrics::counter!("zebra.error.sapling.binding",
+                                                  1,
+                                                  "kind" => binding_sig_error);
+                });
+                // Ignore errors until binding signatures are fixed
+                //.map_err(|e| BoxError::from(Box::new(e)))?;
+            }
+
+            let _rsp = redjubjub_verifier
+                .ready_and()
+                .await?
+                .call((bvk, sapling_shielded_data.binding_sig, &shielded_sighash).into())
+                .boxed();
+
+            // TODO: stop ignoring binding signature errors - #1939
+            // async_checks.push(rsp);
+        }
+
+        // Finally, wait for all asynchronous checks to complete
+        // successfully, or fail verification if they error.
+        while let Some(check) = async_checks.next().await {
+            tracing::trace!(?check, remaining = async_checks.len());
+            check?;
+        }
+
+        Ok(tx.hash())
```

Comment:
The code in this method should be _almost_ identical to what was in the `V4` match arm. It might be easier to check this during review using some sort of automation. The only differences should be related to how `Request` now has getter methods.

Reply:
`git diff --color-moved` works well for me.

It's unfortunate that GitHub doesn't have a similar "move code" review feature.